internal/engine/extractors.go

```diff
@@ -16,3 +19,64 @@ func extractImageAndK8sTargets(specs []model.TargetSpec) (iTargets []model.Image
 	}
 	return iTargets, kTargets
 }
+
+// Check to see if we have k8s targets.
+func extractK8sTargets(specs []model.TargetSpec) []model.K8sTarget {
+	kTargets := make([]model.K8sTarget, 0)
+	for _, spec := range specs {
+		t, ok := spec.(model.K8sTarget)
+		if !ok {
+			continue
+		}
+		kTargets = append(kTargets, t)
+	}
+	return kTargets
+}
+
+// Extract image targets iff they can be updated in-place in a container.
+func extractImageTargetsForLiveUpdates(specs []model.TargetSpec, stateSet store.BuildStateSet) ([]model.ImageTarget, error) {
```

Comment:
are we assuming that intermediate images won't have fast-build instructions? that they won't have `state.DeployInfo` populated?

Reply:
ya, i added some safeguards to make sure that the BuildState will only have DeployInfo populated if the image names match.